Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java

```diff
@@ -125,7 +167,225 @@ public boolean isSkip() {
     }
 
     public String baseIRI() {
-        return baseIRI;
+        return baseIRI.orElse(DEFAULT_BASE_IRI);
     }
 
+    public Iri getRdfType() {
+        return Rdf.iri(rdfType.orElse(DCAT.CATALOG.stringValue()));
+    }
+
+    public String getIri() {
+        return iri;
+    }
+
+    public void addExternalDataset(Dataset dataset) {
+        if (externalDatasets.isEmpty()) {
+            externalDatasets = Optional.of(new ArrayList<>());
+        }
+        externalDatasets.get().add(dataset);
+    }
+
+    public List<Dataset> getExternalDatasets() {
+        return externalDatasets.orElse(Collections.emptyList());
+    }
+
+    public String getQueryStringForCataloging(String newOntopEndpoint) {
+        // makes a sparql query and determine which dataset is already initialised
+        // sets the iri if it is already initialised so that the timestamp can be
+        // modified
+        setDatasetExistsAndIris();
+
+        List<TriplePattern> insertTriples = new ArrayList<>();
+        List<TriplePattern> deleteTriples = new ArrayList<>();
+        List<GraphPattern> wherePatterns = new ArrayList<>();
+
+        StringLiteral currentTime = Rdf.literalOfType(LocalDateTime.now().toString(), XSD.DATETIME);
+
+        SelectQuery query = Queries.SELECT(); // used to generate variable programmatically
+        Variable blazegraphServiceVar = query.var();
+        Variable postgisServiceVar = query.var();
+        Variable geoserverServiceVar = query.var();
+        Variable ontopServiceVar = query.var();
+
+        Iri catalogIri;
+        if (!exists) {
+            iri = SparqlConstants.DEFAULT_NAMESPACE + UUID.randomUUID();
+            catalogIri = Rdf.iri(iri);
+
+            insertTriples.add(catalogIri.isA(getRdfType())
+                    .andHas(DCTERMS.TITLE, getName())
+                    .andHas(DCTERMS.DESCRIPTION, getDescription())
```

Comment:
Would be great to be able to "@" a txt file or something like with other options